### PR TITLE
Move NetX project from .NET ITs to Java ITs

### PR DESCRIPTION
**Pull request by gregory-paidis-sonarsource**
Part of #8921
Moves everything else after `Net5` was merged.

### CODE REVIEW

File: its/src/test/java/com/sonar/it/csharp/SdkTest.java [hunk 1]
```diff
@@ -39,16 +39,36 @@ public class SdkTest {
...
+    validateProject("Net7", 3, 5);
+  }
+
+  @Test
+  void net8() throws IOException {
+    validateProject("Net8", 3, 5);
   }
 
-  void validateProject(String project) throws IOException {
+  void validateProject(String project, int issueLine, int lines) throws IOException {
```
Comment (gregory-paidis-sonarsource) [inline its/src/test/java/com/sonar/it/csharp/SdkTest.java:67]: NetCore31 does not support top level statements, so this needs parametrization.

File: analyzers/its/update-expected.ps1 [hunk 2]
```diff
@@ -2,7 +2,7 @@
 param
 (
     [Parameter(HelpMessage = "The name of single project to update issues for. If ommited, issues for all projects will be updated.")]
-    [ValidateSet("AnalyzeGenerated.CS", "AnalyzeGenerated.VB", "akka.net", "AutoMapper", "BlazorSample", "CSharpLatest", "Ember-MM", "Nancy", "NetCore31", "Net6", "Net7", "Net8", "ManuallyAddedNoncompliantIssues.CS", "ManuallyAddedNoncompliantIssues.VB", "RazorSample", "Roslyn.1.3.1", "SkipGenerated.CS", "SkipGenerated.VB", "SonarLintExclusions", "WebConfig")]
+    [ValidateSet("AnalyzeGenerated.CS", "AnalyzeGenerated.VB", "akka.net", "AutoMapper", "BlazorSample", "CSharpLatest", "Ember-MM", "Nancy", "ManuallyAddedNoncompliantIssues.CS", "ManuallyAddedNoncompliantIssues.VB", "RazorSample", "Roslyn.1.3.1", "SkipGenerated.CS", "SkipGenerated.VB", "SonarLintExclusions", "WebConfig")]
```
Comment (pavel-mikula-sonarsource) [inline analyzers/its/update-expected.ps1:5]: You'll have a rebase conflict with "" here

File: its/src/test/java/com/sonar/it/csharp/SdkTest.java (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply its/src/test/java/com/sonar/it/csharp/SdkTest.java:67]: What about not using top level statements? Make all of the projects DLL with something like
```
public class Sample
{
    public void Method()
    {
        // FIXME: ...
    }
}
```

File: its/src/test/java/com/sonar/it/csharp/SdkTest.java [hunk 3]
```diff
@@ -39,16 +39,36 @@ public class SdkTest {
   @TempDir
   private static Path temp;
 
+  @Test
```
Comment (pavel-mikula-sonarsource) [inline its/src/test/java/com/sonar/it/csharp/SdkTest.java:42]: File `net6.sln` is called `net6.sln` instead of `Net6.sln` like everything else :)

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**